<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="2" passed="0" failed="2" skipped="0">
  <reporter-output>
    <line>
      <![CDATA[Setting up reports and test is getting ready]]>
    </line>
    <line>
      <![CDATA[Setting Done Test Can be started]]>
    </line>
    <line>
      <![CDATA[Trying to start browser and getting application ready]]>
    </line>
    <line>
      <![CDATA[Browser and application is up and running]]>
    </line>
    <line>
      <![CDATA[Test about to end]]>
    </line>
    <line>
      <![CDATA[Test Completed - Report Generated]]>
    </line>
    <line>
      <![CDATA[Web Page launched correctly.. ]]>
    </line>
    <line>
      <![CDATA[Test about to end]]>
    </line>
    <line>
      <![CDATA[Test Completed - Report Generated]]>
    </line>
  </reporter-output>
  <suite started-at="2025-06-21T20:12:57 IST" name="Default suite" finished-at="2025-06-21T20:13:01 IST" duration-ms="4577">
    <groups>
    </groups>
    <test started-at="2025-06-21T20:12:57 IST" name="Default test" finished-at="2025-06-21T20:13:01 IST" duration-ms="4577">
      <class name="com.org.testcases.BookManagementTest">
        <test-method is-config="true" signature="setUpSuite()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:12:57 IST" name="setUpSuite" finished-at="2025-06-21T20:12:57 IST" duration-ms="139" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[Setting up reports and test is getting ready]]>
            </line>
            <line>
              <![CDATA[Setting Done Test Can be started]]>
            </line>
          </reporter-output>
        </test-method> <!-- setUpSuite -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:12:57 IST" name="setup" finished-at="2025-06-21T20:13:01 IST" duration-ms="3634" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[Trying to start browser and getting application ready]]>
            </line>
            <line>
              <![CDATA[Browser and application is up and running]]>
            </line>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="addAuthor()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:13:01 IST" name="addAuthor" finished-at="2025-06-21T20:13:01 IST" duration-ms="55" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [http://3.87.10.224:3000] but found [http://3.87.10.224:3000/]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [http://3.87.10.224:3000] but found [http://3.87.10.224:3000/]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at org.testng.Assert.assertEquals(Assert.java:666)
at com.org.pages.BookManagement.addAuthor(BookManagement.java:57)
at com.org.testcases.BookManagementTest.addAuthor(BookManagementTest.java:51)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addAuthor -->
        <test-method is-config="true" signature="teardownMethod(org.testng.ITestResult)[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:13:01 IST" name="teardownMethod" finished-at="2025-06-21T20:13:01 IST" duration-ms="461" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=addAuthor status=FAILURE method=BookManagementTest.addAuthor()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
            <line>
              <![CDATA[Test about to end]]>
            </line>
            <line>
              <![CDATA[Test Completed - Report Generated]]>
            </line>
          </reporter-output>
        </test-method> <!-- teardownMethod -->
        <test-method signature="addBook()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:13:01 IST" name="addBook" finished-at="2025-06-21T20:13:01 IST" duration-ms="18" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [http://3.87.10.224:3000] but found [http://3.87.10.224:3000/]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [http://3.87.10.224:3000] but found [http://3.87.10.224:3000/]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at org.testng.Assert.assertEquals(Assert.java:666)
at com.org.pages.BookManagement.addBook(BookManagement.java:39)
at com.org.testcases.BookManagementTest.addBook(BookManagementTest.java:25)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[Web Page launched correctly.. ]]>
            </line>
          </reporter-output>
        </test-method> <!-- addBook -->
        <test-method is-config="true" signature="teardownMethod(org.testng.ITestResult)[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:13:01 IST" name="teardownMethod" finished-at="2025-06-21T20:13:01 IST" duration-ms="187" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=addBook status=FAILURE method=BookManagementTest.addBook()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69] output=Web Page launched correctly.. ]]]>
              </value>
            </param>
          </params>
          <reporter-output>
            <line>
              <![CDATA[Test about to end]]>
            </line>
            <line>
              <![CDATA[Test Completed - Report Generated]]>
            </line>
          </reporter-output>
        </test-method> <!-- teardownMethod -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:com.org.testcases.BookManagementTest@662b4c69]" started-at="2025-06-21T20:13:01 IST" name="tearDown" finished-at="2025-06-21T20:13:01 IST" duration-ms="174" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- com.org.testcases.BookManagementTest -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
